systems/subscriber/subscriber-registry/pkg/server/subscriber.go

```diff
@@ -0,0 +1,288 @@
+		return nil, err
+	}
+
+	timestamp := &timestamppb.Timestamp{
+		Seconds: req.DateOfBirth.GetSeconds(),
+		Nanos:   req.DateOfBirth.GetNanos(),
+	}
+
+	birthday := timestamp.AsTime()
+
+	subscriber := &db.Subscriber{
+		OrgID:orgID_uuid,
+		SubscriberID:          subscriberID_uuid,
+		FirstName:             req.GetFirstName(),
+		LastName:              req.GetLastName(),
+		NetworkID:             networkID_uuid,
+		Email:                 req.GetEmail(),
+		PhoneNumber:           req.GetPhoneNumber(),
+		Gender:                req.GetGender(),
+		Address:               req.GetAddress(),
+		ProofOfIdentification: req.GetProofOfIdentification(),
+		DOB:                   birthday,
+		IdSerial:              req.GetIdSerial(),
+	}
+	err = s.subscriberRepo.Add(subscriber)
+	if err != nil {
+		logrus.Error("error while adding subscriber" + err.Error())
+		return nil, grpc.SqlErrorToGrpc(err, "subscriber")
+	}
+
+	return &pb.AddSubscriberResponse{
+		Subscriber: &pb.Subscriber{
+			OrgID: orgID_uuid.String(),
+			SubscriberID:          subscriberID_uuid.String(),
+			FirstName:             req.GetFirstName(),
+			LastName:              req.GetLastName(),
+			NetworkID:             networkID_uuid.String(),
+			Email:                 req.GetEmail(),
+			PhoneNumber:           req.GetPhoneNumber(),
+			Gender:                req.GetGender(),
+			Address:               req.GetAddress(),
+			ProofOfIdentification: req.GetProofOfIdentification(),
+			DateOfBirth:           birthday.GoString(),
+			IdSerial:              req.GetIdSerial()},
+	}, nil
+
+}
+
+func (s *SubcriberServer) Delete(ctx context.Context, req *pb.DeleteSubscriberRequest) (*pb.DeleteSubscriberResponse, error) {
+	subscriberID := req.GetSubscriberID()
+	if subscriberID == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "subscriberID must not be empty")
+	}
+
+	if err := ctx.Err(); err != nil {
+		return nil, err
+	}
+	subscriberIDUUID, err := uuid.FromString(subscriberID)
+
+	logrus.Infof("Delete Subscriber : %v ", subscriberID)
+	err = s.subscriberRepo.Delete(subscriberIDUUID)
+	if err != nil {
+		logrus.WithError(err).Error("error while deleting subscriber")
+		return nil, grpc.SqlErrorToGrpc(err, "subscriber")
+	}
+	route := s.subscriberRoutingKey.SetAction("delete").SetObject("subscriber").MustBuild()
+	err = s.msgbus.PublishRequest(route, req)
+	if err != nil {
+		logrus.Errorf("Failed to publish message %+v with key %+v. Errors %s", req, route, err.Error())
+	}
+	if err != nil {
+		logrus.Errorf("Failed to publish message %+v with key %+v. Errors %s", req, route, err.Error())
+	}
+	return &pb.DeleteSubscriberResponse{
+		SubscriberID: subscriberID,
+	}, nil
+}
+
+func (s *SubcriberServer) Get(ctx context.Context, req *pb.GetSubscriberRequest) (*pb.GetSubscriberResponse, error) {
+
+	subscriberID := req.GetSubscriberID()
+	if subscriberID == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "subscriberID must not be empty")
+	}
+
+	if err := ctx.Err(); err != nil {
```

Comment:
What is that for?

Reply:
This to check the context error .I read about seems like a good practice.I have  added a more in depth desc in the first comment.